examples/linear_deformation.py

```diff
@@ -0,0 +1,1187 @@
+
+    def derivative(self, alphas):
+        """Frechet derivative of this operator in ``alphas``.
+
+        Parameters
+        ----------
+        alphas: `ProductSpaceVector`
+            Deformation parameters for control points. It has ``n``
+            components, each of which has size ``N``. Here, ``n`` is
+            the number of dimensions and ``N`` the number of control
+            points.
+
+        Returns
+        -------
+        deriv : `Operator`
+            The derivative of this operator, evaluated at ``alphas``
+        """
+        deriv_op = DisplacementDerivative(
+            alphas, self.control_points, self.discr_space, self.ft_kernel)
+
+        return deriv_op
+
+
+class DisplacementDerivative(DisplacementOperator):
+
+    """Frechet derivative of the displacement operator at alphas."""
+
+    def __init__(self, alphas, control_points, discr_space, ft_kernel):
+        """Initialize a new instance.
+
+        Parameters
+        ----------
+        alphas : `ProductSpaceVector`
+            Displacement parameters in which the derivative is evaluated
+        control_points : `TensorGrid` or `array-like`
+            The points ``x_j`` controlling the deformation. They can
+            be given either as a tensor grid or as a point array. In
+            the latter case, its shape must be ``(N, n)``, where
+            ``n`` is the dimension of the template space, and ``N``
+            the number of ``alpha_j``, i.e. the size of (each
+            component of) ``par_space``.
+        discr_space : `DiscreteSpace`
+            Space of the image grid of the template.
+        kernel : `callable`
+            Function to determine the kernel at the control points ``K(y_j)``
+            The function must accept a real variable and return a real number.
+        """
+
+        super().__init__(alphas.space, control_points, discr_space, ft_kernel)
+        self.discr_space = discr_space
+        self.range_space = odl.ProductSpace(self.discr_space,
+                                            self.discr_space.ndim)
+        Operator.__init__(self, alphas.space, self.range_space, linear=True)
+        self.alphas = alphas
+
+    def _call(self, betas):
+        """Implementation of ``self(betas)``.
+
+        Parameters
+        ----------
+        betas: `ProductSpaceVector`
+            Deformation parameters for control points. It has ``n``
+            components, each of which has size ``N``. Here, ``n`` is
+            the number of dimensions and ``N`` the number of control
+            points. It should be in the same space as alpha.
+        """
+
+        return self.displacement_ft(betas)
+
+    @property
+    def adjoint(self):
+        """Adjoint of the displacement derivative."""
+        adj_op = DisplacementDerivativeAdjoint(
+            self.alphas, self.control_points, self.discr_space, self.ft_kernel)
+
+        return adj_op
+
+
+class DisplacementDerivativeAdjoint(DisplacementDerivative):
```

Comment:
This class is not needed, it seems like it is a special case of `DisplacementDerivative`, which is itself a special case of `DisplacementOperator`. By some slight rewordings of `DisplacementOperator`, both of these could be removed.